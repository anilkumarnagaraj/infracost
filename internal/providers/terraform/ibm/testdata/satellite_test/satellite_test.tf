terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
      version = "1.42.0-beta0"
    }
  }
}

provider "ibm" {
  # Configuration options
}

provider "aws" {
  region     = "us-east-1"
  access_key = ""
  secret_key = ""
}


module "satellite-aws" {
  #source = "/Users/anilkumarnagaraj/go/src/github.com/IBM-Cloud/infracost/internal/providers/terraform/ibm/testdata/cloudant_test/modules/satellite"
  source = "git::git@github.com:terraform-ibm-modules/terraform-ibm-satellite-aws.git"

  for_each                   = toset(["sat01", "sat02"])
  ibmcloud_api_key           = ""
  aws_region                 = "us-east-1"
  aws_access_key             = ""
  aws_secret_key             = ""
  resource_group             = "default"
  is_location_exist          = false
  location                   = "tf-satellite-${each.key}"
  #location                  = "tf-satellite-01"
  managed_from               = "wdc"
  location_zones             = ["zone-1", "zone-2", "zone-3"]
  host_labels                = ["env:dev"]
  host_provider              = "aws"
  create_cluster             = false
  cluster                    = "cluster01"
  cluster_host_labels        = ["env:dev"]
  create_timeout             = null
  update_timeout             = null
  delete_timeout             = null
}