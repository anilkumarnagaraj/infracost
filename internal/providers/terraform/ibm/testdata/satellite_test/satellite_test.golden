
 Name                                                                              Monthly Qty  Unit   Monthly Cost 
                                                                                                                    
 module.satellite-aws["sat01"].module.satellite-cluster-ec2.aws_instance.this[0]                                    
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat01"].module.satellite-cluster-ec2.aws_instance.this[1]                                    
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat01"].module.satellite-cluster-ec2.aws_instance.this[2]                                    
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat01"].module.satellite-location-ec2.aws_instance.this[0]                                   
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat01"].module.satellite-location-ec2.aws_instance.this[1]                                   
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat01"].module.satellite-location-ec2.aws_instance.this[2]                                   
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat02"].module.satellite-cluster-ec2.aws_instance.this[0]                                    
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat02"].module.satellite-cluster-ec2.aws_instance.this[1]                                    
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat02"].module.satellite-cluster-ec2.aws_instance.this[2]                                    
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat02"].module.satellite-location-ec2.aws_instance.this[0]                                   
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat02"].module.satellite-location-ec2.aws_instance.this[1]                                   
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 module.satellite-aws["sat02"].module.satellite-location-ec2.aws_instance.this[2]                                   
 ├─ Instance usage (Linux/UNIX, on-demand, m5d.xlarge)                                     730  hours       $164.98 
 └─ root_block_device                                                                                               
    └─ Storage (general purpose SSD, gp2)                                                    8  GB            $0.80 
                                                                                                                    
 OVERALL TOTAL                                                                                            $1,989.36 
──────────────────────────────────
60 cloud resources were detected:
∙ 12 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 38 were free:
  ∙ 12 x aws_security_group_rule
  ∙ 6 x aws_route_table_association
  ∙ 6 x aws_subnet
  ∙ 4 x aws_route
  ∙ 2 x aws_internet_gateway
  ∙ 2 x aws_key_pair
  ∙ 2 x aws_route_table
  ∙ 2 x aws_security_group
  ∙ 2 x aws_vpc
∙ 10 are not supported yet, see https://infracost.io/requested-resources:
  ∙ 6 x ibm_satellite_host
  ∙ 2 x aws_placement_group
  ∙ 2 x ibm_satellite_location